Unit tests are failing

Latest master:
```
> Task :app:testProdDebugUnitTest

fr.free.nrw.commons.leaderboard.LeaderboardFragmentUnitTests > testMenuVisibilityOverrideVisibleWithContext FAILED
    java.lang.Exception at AndroidTestEnvironment.java:538
        Caused by: java.lang.NullPointerException at LeaderboardFragmentUnitTests.kt:232

fr.free.nrw.commons.profile.achievements.AchievementsFragmentUnitTests > testMenuVisibilityOverrideVisibleWithContext FAILED
    java.lang.Exception at AndroidTestEnvironment.java:538
        Caused by: java.lang.NullPointerException at AchievementsFragmentUnitTests.kt:375

fr.free.nrw.commons.customselector.ui.selector.ImageFragmentTest > testHandleResult FAILED
    java.lang.Exception at AndroidTestEnvironment.java:538
        Caused by: java.lang.reflect.InvocationTargetException at ImageFragmentTest.kt:128
            Caused by: java.lang.NullPointerException at ImageFragmentTest.kt:128

fr.free.nrw.commons.customselector.ui.selector.FolderFragmentTest > testHandleResult FAILED
    java.lang.Exception at AndroidTestEnvironment.java:538
        Caused by: java.lang.reflect.InvocationTargetException at FolderFragmentTest.kt:128
            Caused by: java.lang.NullPointerException at FolderFragmentTest.kt:128
```

@nicolas-raoul i have already solved a part of this issue MUCH BEFORE , merge this pull request #4884

fr.free.nrw.commons.customselector.ui.selector.ImageFragmentTest > testHandleResult FAILED
    java.lang.Exception at AndroidTestEnvironment.java:538
        Caused by: java.lang.reflect.InvocationTargetException at ImageFragmentTest.kt:128
            Caused by: java.lang.NullPointerException at ImageFragmentTest.kt:128

fr.free.nrw.commons.customselector.ui.selector.FolderFragmentTest > testHandleResult FAILED
    java.lang.Exception at AndroidTestEnvironment.java:538
        Caused by: java.lang.reflect.InvocationTargetException at FolderFragmentTest.kt:128
            Caused by: java.lang.NullPointerException at FolderFragmentTest.kt:128

and this two test fail will be gone

Thanks @Rishavgupta12345, now the remaining failing unit tests are:
```
fr.free.nrw.commons.leaderboard.LeaderboardFragmentUnitTests > testMenuVisibilityOverrideVisibleWithContext FAILED
    java.lang.Exception at AndroidTestEnvironment.java:538
        Caused by: java.lang.NullPointerException at LeaderboardFragmentUnitTests.kt:232

fr.free.nrw.commons.profile.achievements.AchievementsFragmentUnitTests > testMenuVisibilityOverrideVisibleWithContext FAILED
    java.lang.Exception at AndroidTestEnvironment.java:538
        Caused by: java.lang.NullPointerException at AchievementsFragmentUnitTests.kt:375
```

I just ran `./gradlew clean testProdDebugUnitTest` in master (which has #4884 merged in already), it still fails with:
```
fr.free.nrw.commons.leaderboard.LeaderboardFragmentUnitTests > testMenuVisibilityOverrideVisibleWithContext FAILED
    java.lang.Exception at AndroidTestEnvironment.java:538
        Caused by: java.lang.NullPointerException at LeaderboardFragmentUnitTests.kt:232

fr.free.nrw.commons.profile.achievements.AchievementsFragmentUnitTests > testMenuVisibilityOverrideVisibleWithContext FAILED
    java.lang.Exception at AndroidTestEnvironment.java:538
        Caused by: java.lang.NullPointerException at AchievementsFragmentUnitTests.kt:375
```

Anyone who finds a fix, please feel free to send a pull request!
Since this is quite urgent, anyone please feel free to work on this particular even if not assigned.
Thanks in advance!

--- a/app/src/test/kotlin/fr/free/nrw/commons/leaderboard/LeaderboardFragmentUnitTests.kt
+++ b/app/src/test/kotlin/fr/free/nrw/commons/leaderboard/LeaderboardFragmentUnitTests.kt
@@ -2,6 +2,7 @@ package fr.free.nrw.commons.leaderboard
 
 import android.accounts.Account
 import android.content.Context
+import android.os.Looper.getMainLooper
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
@@ -31,6 +32,8 @@ import org.powermock.reflect.Whitebox
 import org.robolectric.Robolectric
 import org.robolectric.RobolectricTestRunner
 import org.robolectric.RuntimeEnvironment
+import org.robolectric.Shadows
+import org.robolectric.Shadows.shadowOf
 import org.robolectric.annotation.Config
 import org.robolectric.annotation.LooperMode
 import org.robolectric.shadows.ShadowToast
@@ -222,6 +225,7 @@ class LeaderboardFragmentUnitTests {
     @Test
     @Throws(Exception::class)
     fun testMenuVisibilityOverrideVisibleWithContext() {
+        shadowOf(getMainLooper()).idle()
         `when`(parentView.context).thenReturn(context)
         val method: Method = LeaderboardFragment::class.java.getDeclaredMethod(
             "setMenuVisibility",
@@ -229,7 +233,9 @@ class LeaderboardFragmentUnitTests {
         )
         method.isAccessible = true
         method.invoke(fragment, true)
-        Assert.assertEquals(ShadowToast.getTextOfLatestToast().toString(), context.getString(R.string.leaderboard_unavailable_beta))
+        Assert.assertNotNull(ShadowToast.getTextOfLatestToast().toString())
+        Assert.assertEquals(ShadowToast.getTextOfLatestToast().toString()
+            , context.getString(R.string.leaderboard_unavailable_beta))
     }
 
 }

--- a/app/src/test/kotlin/fr/free/nrw/commons/leaderboard/LeaderboardFragmentUnitTests.kt
+++ b/app/src/test/kotlin/fr/free/nrw/commons/leaderboard/LeaderboardFragmentUnitTests.kt
@@ -27,6 +27,7 @@ import org.junit.Test
 import org.junit.runner.RunWith
 import org.mockito.Mock
 import org.mockito.Mockito.`when`
+import org.mockito.Mockito.doReturn
 import org.mockito.MockitoAnnotations
 import org.powermock.reflect.Whitebox
 import org.robolectric.Robolectric
@@ -233,9 +234,17 @@ class LeaderboardFragmentUnitTests {
         )
         method.isAccessible = true
         method.invoke(fragment, true)
-        Assert.assertNotNull(ShadowToast.getTextOfLatestToast().toString())
-        Assert.assertEquals(ShadowToast.getTextOfLatestToast().toString()
-            , context.getString(R.string.leaderboard_unavailable_beta))
+        if(isBetaFlavour) {
+            Assert.assertEquals(
+                ShadowToast.getTextOfLatestToast().toString(),
+                context.getString(R.string.leaderboard_unavailable_beta)
+            )
+        } else {
+            Assert.assertNull(
+                ShadowToast.getTextOfLatestToast()
+            )
+        }
+
     }
 
 }

--- a/app/src/test/kotlin/fr/free/nrw/commons/profile/achievements/AchievementsFragmentUnitTests.kt
+++ b/app/src/test/kotlin/fr/free/nrw/commons/profile/achievements/AchievementsFragmentUnitTests.kt
@@ -17,6 +17,7 @@ import fr.free.nrw.commons.TestAppAdapter
 import fr.free.nrw.commons.TestCommonsApplication
 import fr.free.nrw.commons.auth.SessionManager
 import fr.free.nrw.commons.profile.ProfileActivity
+import fr.free.nrw.commons.utils.ConfigUtils
 import org.junit.Assert
 import org.junit.Before
 import org.junit.Test
@@ -372,7 +373,16 @@ class AchievementsFragmentUnitTests {
         )
         method.isAccessible = true
         method.invoke(fragment, true)
-        Assert.assertEquals(ShadowToast.getTextOfLatestToast().toString(), context.getString(R.string.achievements_unavailable_beta))
+        if(ConfigUtils.isBetaFlavour) {
+            Assert.assertEquals(
+                ShadowToast.getTextOfLatestToast().toString(),
+                context.getString(R.string.achievements_unavailable_beta)
+            )
+        } else {
+            Assert.assertNull(
+                ShadowToast.getTextOfLatestToast()
+            )
+        }
     }
 
 }

